packages/core/core-flows/src/product/workflows/update-products.ts

```diff
@@ -20,7 +37,87 @@ export const updateProductsWorkflow = createWorkflow(
     input: WorkflowData<WorkflowInput>
   ): WorkflowData<ProductTypes.ProductDTO[]> => {
     // TODO: Delete price sets for removed variants
-    // TODO Update sales channel links
-    return updateProductsStep(input)
+
+    const toUpdateInput = transform({ input }, (data) => {
+      if ((data.input as UpdateProductsStepInputProducts).products) {
+        return (data.input as UpdateProductsStepInputProducts).products.map(
+          (p) => ({
+            ...p,
+            sales_channels: undefined,
+          })
+        )
+      } else {
+        return {
+          selector: (data.input as UpdateProductsStepInputSelector).selector,
+          update: {
+            ...(data.input as UpdateProductsStepInputSelector).update,
+            sales_channels: undefined,
+          },
+        }
+      }
+    })
+
+    const updatedProducts = updateProductsStep(toUpdateInput as any) // TODO: type
+
+    const updatedProductIds = transform({ updatedProducts, input }, (data) => {
+      if (
+        !(data.input as UpdateProductsStepInputSelector).update.sales_channels
+      ) {
+        return []
+      }
+
+      let productIds = data.updatedProducts.map((p) => p.id)
+
+      ;(data.input as UpdateProductsStepInputProducts).products?.forEach(
+        (p) => {
+          if (!p.sales_channels) {
+            // these products don't update sales channels so we don't want to their previous channels
+            productIds = productIds.filter((id) => id !== p.id)
+          }
+        }
+      )
+
+      return productIds
+    })
+
+    const currentLinks = getProductSalesChannelLinkStep({
+      ids: updatedProductIds,
+    })
```

Comment:
no need to create `getProductSalesChannelLinkStep` function. you can just reuse the commom step like this:

```
useRemoteQueryStep({
  entry_point: "product_sales_channel",
  fields: ["product_id", "sales_channel_id"],
  variables: { product_id: updatedProductIds },
});
```